pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -90,13 +90,19 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 	if err != nil {
 		err = errors.Wrapf(err, "error creating bucket %q", p.bucketName)
 		logger.Errorf(err.Error())
-		p.deleteOBCResource("")
+		if err := p.deleteOBCResource(""); err != nil {
+			err = errors.Wrap(err, "error deleting OBCResource bucket")
+			return nil, err
+		}
 		return nil, err
 	}
 
 	_, errCode, err := cephObject.SetQuotaUserBucketMax(p.objectContext, p.cephUserName, maxBuckets)
 	if errCode > 0 {
-		p.deleteOBCResource(p.bucketName)
+		if err := p.deleteOBCResource(p.bucketName); err != nil {
+			err = errors.Wrapf(err, "error deleting OBCResource bucket %q", p.bucketName)
```

Comment:
It's simpler to return on the same line, looks like there are other places in this same file as well.
```suggestion
			return nil, errors.Wrapf(err, "error deleting OBCResource bucket %q", p.bucketName)
```

Reply:
Done.